### ISSUE TITLE
implementing stochastic inference for sparse GP using StreamingDenseFeatures

### ISSUE THREAD

**Issue by yorkerlin** [shogun-toolbox/shogun#3173]
reference:
Gaussian Process for Big Data
Scalable Variational Gaussian Process Classification

**Comment by yorkerlin** [comment]
@karlnapf 
The idea is to adapt `vowpal wabbit` as a distributed stochastic minimizer.
The minimizer will be used in sparse GP.
My goal is to extract useful lines of code from the recent `vw` and import them to Shogun's optimization.

The `vw` code in shogun is from vw `5.1`. It is old.
Note that this issue is not about updating the `vw` modular in shogun. 

Road map (`start from something you are familiar with`) :
* understanding/reading shogun's vw modular
* understanding vw `5.1` which is used in shogun's vw modular
* extract useful line of code from vw `7.10`


any suggestion?

**Comment by karlnapf** [comment]
Yes, this update would be very useful. I don't know who did the VW port back then.
But you can definitely talk to John Langford, who is interestewd in these bindings to work. cc me in if you do

**Comment by karlnapf** [comment]
And yes, we should use it as an optimizer

**Comment by yorkerlin** [comment]
I am working this from now on.
I will comment some existing codes in shogun's VW modular.
